test/CollapseTable.js

```diff
@@ -48,4 +48,104 @@ describe('CollapseTable', () => {
       assert.deepEqual(actual, [])
     })
   })
+
+  describe('.toggleCollapseClickCallback()', () => {
+    const toggleCollapseClickCallback = applib.CollapseTable.toggleCollapseClickCallback
+
+    describe('and an open container', () => {
+      beforeEach(function Test() {
+        const html = `<div>
+                        <div
+                          id=header
+                          class='app_table_collapse_close app_table_collapse_icon'
+                          style='visibility: visible'>
+                          <span
+                            id=caption
+                            class=app_table_collapsed_caption
+                            style='visibility: hidden'
+                          </span>
+                        </div>
+                        <table style='display: block'></table>
+                        <div id=footer style='display: block'></div>
+                      </div>`
+        this.doc = domino.createDocument(html)
+      })
+
+      it('where the footer is observed, the callback is invoked when the footer is clicked',
```

Comment:
I tried a more descriptive style for each test case. Let me know your thoughts on this and I'll make the tests consistent one way or another.